storage/tests/unit/test_blob.py

```diff
@@ -459,6 +492,26 @@ def test__get_download_url_on_the_fly_with_user_project(self):
                 user_project))
         self.assertEqual(download_url, expected_url)
 
+    def test__get_download_url_on_the_fly_with_kms_key_name(self):
+        from six.moves.urllib.parse import urlencode
+        kms_resource = (
+            "projects/test-project-123/"
+            "locations/global/"
+            "keyRings/test-ring/"
+            "cryptoKeys/test-key/"
+        )
+        blob_name = 'bzzz-fly.txt'
+        bucket = _Bucket(name='buhkit')
+        blob = self._make_one(
+            blob_name, bucket=bucket, kms_key_name=kms_resource)
+
+        self.assertIsNone(blob.media_link)
+        download_url = blob._get_download_url()
+        expected_url = (
+            'https://www.googleapis.com/download/storage/v1/b/'
+            'buhkit/o/bzzz-fly.txt?alt=media')  # kmsKeyName *not* expected
```

Comment:
That's correct. KMS key name associated to the object will be used on a GET request. The prerequisites are in the one-pager.